Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp

```diff
@@ -415,17 +495,26 @@ void IntegratePeaksMD2::integrate(typename MDEventWorkspace<MDE, nd>::sptr ws) {
             bgSignal / (4 * M_PI * pow(PeakRadiusVector[i], 3) / 3);
         std::vector<V3D> eigenvects;
         std::vector<double> eigenvals;
-        findEllipsoid<MDE, nd>(
-            ws, getRadiusSq, pos,
-            static_cast<coord_t>(pow(PeakRadiusVector[i], 2)), qAxisIsFixed,
-            bgDensity, eigenvects, eigenvals);
+        if (PeakRadius.size() == 1) {
+          findEllipsoid<MDE, nd>(
+              ws, getRadiusSq, pos,
+              static_cast<coord_t>(pow(PeakRadiusVector[i], 2)), qAxisIsFixed,
+              bgDensity, eigenvects, eigenvals);
+        } else {
+          // Use the manually specified radii instead of finding them via
+          // findEllipsoid
+          eigenvals = PeakRadius;
```

Comment:
The `eigenvals` should be the radii squared.


```diff
--- a/Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp
+++ b/Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp
@@ -503,7 +503,9 @@ void IntegratePeaksMD2::integrate(typename MDEventWorkspace<MDE, nd>::sptr ws) {
         } else {
           // Use the manually specified radii instead of finding them via
           // findEllipsoid
-          eigenvals = PeakRadius;
+          eigenvals.push_back(pow(PeakRadius[0], 2));
+          eigenvals.push_back(pow(PeakRadius[1], 2));
+          eigenvals.push_back(pow(PeakRadius[2], 2));
           eigenvects.push_back(V3D(1.0, 0.0, 0.0));
           eigenvects.push_back(V3D(0.0, 1.0, 0.0));
           eigenvects.push_back(V3D(0.0, 0.0, 1.0));
```